client/mysqldump.c

```diff
@@ -6250,43 +6264,29 @@ static int do_show_slave_status(MYSQL *mysql_con, int have_mariadb_gtid,
 
 static int do_start_slave_sql(MYSQL *mysql_con)
 {
-  MYSQL_RES *slave;
-  MYSQL_ROW row;
   int error= 0;
   DBUG_ENTER("do_start_slave_sql");
-
-  /* We need to check if the slave sql is stopped in the first place */
-  if (mysql_query_with_error_report(mysql_con, &slave,
-                                    multi_source ?
-                                    "SHOW ALL SLAVES STATUS" :
-                                    "SHOW SLAVE STATUS"))
-    DBUG_RETURN(1);
-
-  while ((row= mysql_fetch_row(slave)))
+  for (; n_stopped_replicas--;)
   {
-    DBUG_PRINT("info", ("Connection: '%s'  status: '%s'",
-                        multi_source ? row[0] : "", row[11 + multi_source]));
-    if (row[11 + multi_source])
+    /*
+      do_start_slave_sql() should only be called
+      sometime after do_stop_slave_sql() suceeds
+    */
+    char* stopped_replica= stopped_replicas[n_stopped_replicas];
+    char query[sizeof("START SLAVE ''") + NAME_CHAR_LEN];
+    DBUG_PRINT("info", ("Connection: '%.*s'", NAME_CHAR_LEN, stopped_replica));
+    // if SLAVE SQL is running, start it anyway to warn unexpected state change
```

Comment:
The program runs SHOW SLAVE STATUS multiple times thoughout the various step functions:
* `do_show_slave_status`
* `do_stop_slave_sql`
* `do_start_slave_sql` (before this PR)

Perhaps the intent is in case replication configs change in the middle of `mariadb-dump`?